### PR TITLE
refactor: Move frontend for RHACM into a separate package 

### PR DESCRIPTION
**Pull request by tumido**
Shuffle code so we can have a standalone RHACM frontend package

Depends on #160

Exposes:

- "Clusters page" as a `RhacmPage` plugin default page and as `ClusterStatusPage` component
- Cluster context as `useCluster` and `ClusterContextProvider`
- Additional UI components as building blogs: `ClusterAllocatableResourceCard`, `ClusterAvailableResourceCard`, `ClusterInfoCard`, `ClusterStatusCard`

Creates an additional package `rhacm-common` which carries type/API spec for both frontend and backend.

Most of the code is just being moved around, I've tried to limit code changes to minimum. The only major thing I've changed is that I've made `TableCardFromData` component from `tableFromObject` function.

This PR should not introduce any visible changes.

### CODE REVIEW

**Review by SamoKopecky — COMMENTED** [review]
Two small nits, a question about the tests. otherwise looks good!

**Comment by SamoKopecky** [comment]
I noticed some small things in my previous code that could be done better, so I'll create a PR for them after this is merged.

File: plugins/rhacm/src/components/ExampleFetchComponent/ExampleFetchComponent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (tumido) [reply plugins/rhacm/src/components/ExampleFetchComponent/ExampleFetchComponent.test.tsx:1]: This is a left over from the boilerplate.. I didn't mean to include `Example*Component`s into this PR. Will be removed.

File: plugins/rhacm/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# rhacm
```
[the remark this answers is not in the document]
Comment (tumido) [reply plugins/rhacm/README.md:1]: Let's leave this as it is for now. I have further refectoring in the pipeline, so we will have to redo all the documentation anyways...

**Review by SamoKopecky — APPROVED** [review]
Looks really good now that it has proper structure!
/lgtm
/approve

**Pull request merged**